azure-pipelines/main-pipeline.yml

```diff
@@ -253,15 +263,100 @@ stages:
               # Sonar Cloud Analysis
 
               - task: SonarCloudAnalyze@1
+                displayName: Analyze
 
               - task: SonarCloudPublish@1
                 inputs:
                   pollingTimeoutSec: "300"
+                displayName: Publish Sonar Results
+
+              # Docker Image is created and pushed to ACR with build specific tag.
+
+              - template: "./azure-templates/docker-template.yml"
+                parameters:
+                  repository: ${{ service.name }}
+                  tag: "$(Build.BuildNumber)"
+                  Dockerfile: ./${{ service.name }}/Dockerfile
+                  containerRegistry: "acr-team3"
+              
+              # Docker Image is created and pushed to ACR with latest tag.
+
+              - template: "./azure-templates/docker-template.yml"
+                parameters:
+                  repository: ${{ service.name }}
+                  tag: "latest"
+                  Dockerfile: ./${{ service.name }}/Dockerfile
+                  containerRegistry: "acr-team3"
+
+  # Deployment Stage. Using images pushed to the Azure Container Registry, and a helm Chart.
+
+  - stage: Deploy
+    displayName: Deploy to AKS Cluster
+    dependsOn: Check
+    condition: and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))
+    jobs:
+      - deployment: DevDeploy
+        displayName: Deploy to AKS Dev Environment
+        condition: eq(variables['Build.SourceBranch'], 'refs/heads/development')
+        environment: dev
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+
+            # Helm deploys to a cluster, dependant on branch. development = dev, staging = staging (main = production)
+              
+              # Download Repo
 
-              # Publish Artifacts to be used in Docker Stage
+              - checkout: self
 
-              - task: PublishPipelineArtifact@1
+              # Creating Image Pull secrets for Helm/K8S to access the images and use on the cluster.
+
+              - task: KubernetesManifest@0
+                displayName: Create Pull Secret Dev
+                inputs:
+                  action: "createSecret"
+                  kubernetesServiceConnection: "k8s-team3-config-dev"
+                  namespace: "team3"
+                  secretType: "dockerRegistry"
+                  secretName: "team3pullsecret"
+                  dockerRegistryEndpoint: "acr-team3"
+
+              # Helm deploys to a cluster, dependant on branch. development = dev, staging = staging (main = production)
+
+              - template: "./azure-templates/helm-template.yml"
```

Comment:
these templates should be better documented in their files, to indicate pre-conditions like "there must be a chart in `./swag-chart`".

Reply:
Resolved in upcoming pr.